contracts/deploy/007_upgrade_single_asset_staking.js

```diff
@@ -0,0 +1,82 @@
+//
+// Script to upgrade the Single Asset Staking contract.
+//
+const {
+  getAssetAddresses,
+  isMainnet,
+  isRinkeby,
+  isFork,
+  isTest,
+  isMainnetOrRinkebyOrFork,
+} = require("../test/helpers.js");
+const { utils } = require("ethers");
+const {
+  log,
+  deployWithConfirmation,
+  executeProposal,
+} = require("../utils/deploy");
+const { proposeArgs } = require("../utils/governor");
+
+const deployName = "007_upgrade_single_asset_staking";
+
+const upgradeSingleAssetStaking = async ({ getNamedAccounts, deployments }) => {
+  console.log(`Running ${deployName} deployment...`);
+
+  const { governorAddr, deployerAddr } = await getNamedAccounts();
+
+  const assetAddresses = await getAssetAddresses(deployments);
+
+  const sDeployer = ethers.provider.getSigner(deployerAddr);
+  const sGovernor = ethers.provider.getSigner(governorAddr);
+
+  //
+  // Deploy the new implementation.
+  //
+  const dSingleAssetStaking = await deployWithConfirmation(
+    "SingleAssetStaking"
+  );
+
+  //
+  // Upgrade.
+  //
+  const cOGNStakingProxy = await ethers.getContract("OGNStakingProxy");
```

Comment:
@franckc not sure do we need to call initialise like here or it is not needed?
https://github.com/OriginProtocol/origin-dollar/blob/master/contracts/deploy/004_single_asset_staking.js#L96-L98

Reply:
When using proxies, the proxy is used for storing all the data. Since we initialized this data already during the 004 deploy, we don't need to do it again. we are in effect just upgrading the logic.